winlog.go

```diff
@@ -97,17 +110,20 @@ func (self *WinLogWatcher) subscribeWithoutBookmark(channel, query string, flags
 	if err != nil {
 		return fmt.Errorf("Failed to create new bookmark handle: %v", err)
 	}
-	callback := &LogEventCallbackWrapper{callback: self, subscribedChannel: channel}
+	callback := newEventCallbackWrapper(self, channel)
 	subscription, err := CreateListener(channel, query, flags, callback)
 	if err != nil {
 		CloseEventHandle(uint64(newBookmark))
 		return err
 	}
+	wrappersMutex.RLock()
 	self.watches[channel] = &channelWatcher{
+		wrapperKey:   callback,
 		bookmark:     newBookmark,
 		subscription: subscription,
-		callback:     callback,
+		callback:     wrappers[callback],
```

Comment:
Move global map access to single unit should simplify understanding of Lock-Unlock chain.

Should we a add simple `getWrapper(key)` helper with `RLock+RUnlock` logic inside?